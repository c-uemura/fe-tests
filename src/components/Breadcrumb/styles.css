.breadcrumb {
  font-size: 12px;
  position: absolute;
  top: 60px;
  left: 50px;
}
