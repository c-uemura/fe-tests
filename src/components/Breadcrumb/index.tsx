import { useLocation } from 'react-router-dom';
import './styles.css';

const Breadcrumb = () => {
  const { pathname } = useLocation();
  const name = pathname.replace(/\//, '');
  return <div className="breadcrumb">{name === '' ? 'start page' : name}</div>;
};

export default Breadcrumb;
