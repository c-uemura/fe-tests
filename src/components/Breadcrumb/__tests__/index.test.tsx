import { vi } from 'vitest';
import { render } from '@testing-library/react';
import Breadcrumb from '@/components/Breadcrumb';

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...(actual as any),
    useLocation: () => vi.fn(),
  };
});

describe('Breadcrumb', () => {
  it('render element', async () => {
    const router = await import('react-router-dom');
    router.useLocation = vi.fn().mockReturnValue({ pathname: '/foo' });

    const { baseElement } = render(<Breadcrumb />);
    expect(baseElement).toMatchSnapshot();
  });

  it('render start page', async () => {
    const router = await import('react-router-dom');
    router.useLocation = vi.fn().mockReturnValue({ pathname: '/' });

    const { baseElement } = render(<Breadcrumb />);
    expect(baseElement).toMatchSnapshot();
  });
});
