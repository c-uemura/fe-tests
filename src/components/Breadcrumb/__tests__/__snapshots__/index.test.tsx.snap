// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb > render element 1`] = `
<body>
  <div>
    <div
      class="breadcrumb"
    >
      foo
    </div>
  </div>
</body>
`;

exports[`Breadcrumb > render start page 1`] = `
<body>
  <div>
    <div
      class="breadcrumb"
    >
      start page
    </div>
  </div>
</body>
`;
